File: src/console/log/logger/ILogger.php
<?php

namespace pocketcloud\cloud\console\log\logger;

use pocketcloud\cloud\console\log\level\CloudLogLevel;
use Throwable;

interface ILogger {

    public function info(string $message, string ...$params): self;

    public function warn(string $message, string ...$params): self;

    public function error(string $message, string ...$params): self;

    public function success(string $message, string ...$params): self;

    public function debug(string $message, string ...$params): self;

    public function forceDebug(string $message, string ...$params): self;

    public function exception(Throwable $throwable): self;

    public function log(CloudLogLevel $logLevel, string $message, mixed... $params): self;

    public function emptyLine(bool $prefix = false, ?CloudLogLevel $logLevel = null): self;

    public function echo(string $message): void;

    public function dump(mixed ...$vars): void;

    public function close(): void;

    public function setFormat(?string $format): self;

    public function resetFormat(): self;

    public function getFormat(): ?string;

    public function setDebugMode(bool $enabled): void;

    public function isDebugMode(): bool;

    public function setSaveLogs(bool $enabled): void;

    public function isSaveLogs(): bool;
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/console/log/ILogger.java
package de.pocketcloud.cloud.console.log;

public interface ILogger {

    default ILogger info(String message, Object... params) {
        return log(CloudLogLevel.INFO, message, params);
    }

    default ILogger warn(String message, Object... params) {
        return log(CloudLogLevel.WARN, message, params);
    }

    default ILogger error(String message, Object... params) {
        return log(CloudLogLevel.ERROR, message, params);
    }

    default ILogger success(String message, Object... params) {
        return log(CloudLogLevel.SUCCESS, message, params);
    }

    default ILogger debug(String message, Object... params) {
        return log(CloudLogLevel.DEBUG, message, params);
    }

    ILogger exception(Throwable throwable);

    ILogger exception(String message, Throwable throwable, Object... params);

    ILogger log(CloudLogLevel logLevel, String message, Object... params);

    ILogger withoutFormat(String message, Object... params);

    default ILogger emptyLine() {
        return echo("");
    }

    ILogger echo(String message);

    ILogger setDebugMode(boolean debugMode);

    boolean isDebugMode();

    ILogger setSaveLogs(boolean enabled);

    boolean isSaveLogs();
}